File: qe-selenium-java/src/main/java/demos/ActivityFifteen.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class ActivityFifteen {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-attributes");
        driver.manage().window().maximize();

        System.out.println(driver.getTitle());

        driver.findElement(By.xpath("//input[starts-with(@id,'full-name')]"))
                .sendKeys("Milind Nautiyal");

        driver.findElement(By.xpath("//input[contains(@id,'email')]"))
                .sendKeys("milind.nautiyal@gmail.com");

        driver.findElement(By.xpath("//input[@type='date']"))
                .sendKeys("2003-01-20"); // correct format for date input

        driver.findElement(By.xpath("//textarea[contains(@id,'additional-details')]"))
                .sendKeys("hello, my name is Milind.");

        driver.findElement(By.xpath("//button[text()='Submit']")).click();

        WebElement confirmation = wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                        By.xpath("//h3[@id='action-confirmation']")));

        System.out.println("Confirmation message: " + confirmation.getText());

        driver.quit();
    }
}

File: qe-selenium-python/Activity_Fifteen.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-attributes")
driver.maximize_window()

print(driver.title)

driver.find_element(By.XPATH, "//input[starts-with(@id,'full-name')]") \
    .send_keys("Milind Nautiyal")

driver.find_element(By.XPATH, "//input[contains(@id,'email')]") \
    .send_keys("milind.nautiyal@gmail.com")

driver.find_element(By.XPATH, "//input[@type='date']") \
    .send_keys("2003-01-20")

driver.find_element(By.XPATH, "//textarea[contains(@id,'additional-details')]") \
    .send_keys("hello, my name is Milind.")

driver.find_element(By.XPATH, "//button[text()='Submit']").click()

confirmation = wait.until(
    EC.visibility_of_element_located((By.XPATH, "//h3[@id='action-confirmation']"))
)

print("Confirmation message:", confirmation.text)

driver.quit()
